stan/math/rev/mat/fun/scale_matrix_exp_multiply.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef STAN_MATH_REV_MAT_FUN_SCALE_MATRIX_EXP_MULTIPLY_HPP
+#define STAN_MATH_REV_MAT_FUN_SCALE_MATRIX_EXP_MULTIPLY_HPP
+
+#include <stan/math/rev/mat/fun/matrix_exp_multiply.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <boost/math/tools/promotion.hpp>
+#include <boost/utility/enable_if.hpp>
+#include <boost/type_traits.hpp>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/**
+ * Wrapper of matrix_exp_action function for a more literal name.
+ * @tparam Ta scalar type matrix A
+ * @tparam N Rows and cols matrix A, also rows of matrix B
+ * @tparam Tb scalar type matrix B
+ * @tparam Cb Columns matrix B
+ * @param[in] A Matrix
+ * @param[in] B Matrix
+ * @param[in] t double
+ * @return exponential of At multiplies B
+ */
+template <typename Ta, int N, typename Tb, int Cb>
+inline typename boost::enable_if_c<boost::is_same<Ta, var>::value
```

Comment:
[optional]

We'd like to move to the `std::` version of these.  

[also optional]

Also, we haven't been particularly concerned about tightly constraining input types if not constraining them does not lead to ambiguities in the instantiations that we care about.  So if this isn't here to remove ambiguity, feel free to remove all the enabling logic.

Reply:
removed `enable_if_c`
```c++
template <typename Ta, typename Tb, int Cb>
inline Eigen::Matrix<typename stan::return_type<Ta, Tb>::type, -1, Cb>
scale_matrix_exp_multiply(const double& t, const Eigen::Matrix<Ta, -1, -1>& A,
                          const Eigen::Matrix<Tb, -1, Cb>& B) {
```